syntax = "proto3";

import "google/protobuf/timestamp.proto";

// ArticleDetails has the same contents as request from client with timestamp
// converted to protobuf format.
// Also same contents as database PostsEntry without global_id
message ArticleDetails {
  string author = 1;
  string title = 2;
  string body = 3;
  google.protobuf.Timestamp creation_datetime = 4;
}

// NewForeignArticle
// converted to protobuf format.
// Also same contents as database PostsEntry without global_id
message NewForeignArticle {
  string attributedTo = 1;
  string title = 2;
  string body = 3;
  string recipient = 4;
  google.protobuf.Timestamp published = 5;
}

// CreateResponse is a simple response.
message CreateResponse {
    enum ResultType {
        OK = 0;
        ERROR = 1;
    }

    ResultType result_type = 1;

    /* Should only be set if result_type is not OK. */
    string error = 2;
}

service Create {
  rpc SendCreate(ArticleDetails) returns (CreateResponse);
  rpc ReceiveCreate(NewForeignArticle) returns (CreateResponse);
}
